lms/djangoapps/grades/tests/test_subsection_grade_factory.py

```diff
@@ -134,13 +141,18 @@ def test_update_with_override(self):
         PersistentSubsectionGradeOverride.update_or_create_override(
             UserFactory(),
             persistent_grade,
-            earned_graded_override=0,
+            earned_graded_override=graded_earned_override,
             earned_all_override=0,
+            possible_graded_override=graded_possible_override,
             feature=PersistentSubsectionGradeOverrideHistory.GRADEBOOK,
         )
 
         # Now, even if the problem scores interface gives us a 2/3,
         # the subsection grade returned should be 0/3 due to the override.
         with mock_get_score(2, 3):
             grade = self.subsection_grade_factory.update(self.sequence)
-            self.assert_grade(grade, 0, 3)
+            self.assert_grade(
```

Comment:
Ah, I understand the issue now.  You're explicitly passing nulls into `update_or_create_override()`.  That function will pull default values from the persistent subsection grade object for a field if that field is not specified, but you are specifying it - it's just null.  I would change this code as follows:
* Change the parameter names to `earned_graded_override` and `possible_graded_override`
* Change the assertion to:
```python
if earned_graded_override is not None:
    self.assert_grade(grade, 0, 3)
else:
    self.assert_grade(grade, 2, 3) # we made an override of possible points only
```
* Do this to create the override above:
```python
kwargs = {
    'feature': PersistentSubsectionGradeOverrideHistory.GRADEBOOK,
}
if earned_graded_override is not None:
    kwargs['earned_graded_override'] = earned_graded_override
if possible_graded_override is not None:
    kwargs['possible_graded_override'] = possible_graded_override
PersistentSubsectionGradeOverride.update_or_create_override(
    UserFactory(),
    persistent_grade,
    **kwargs,
}
```

Reply:
@iloveagent57 With the tests here I want to make sure, if we ever have the case where in the override data, the possible_graded_override OR earned_graded_override columns have explicit value of NULL, we should default back to PersistentSubsectionGrade values. Your suggestion around kwargs are not what I was trying to test for.